workers/generic-worker/process/process.go

```diff
@@ -7,29 +7,41 @@ import (
 	"sync"
 	"syscall"
 	"time"
+
+	"github.com/shirou/gopsutil/v4/mem"
 )
 
 func (r *Result) Succeeded() bool {
 	return r.SystemError == nil && r.ExitError == nil && !r.Aborted
 }
 
-type Command struct {
-	mutex sync.RWMutex
-	*exec.Cmd
-	// abort channel is closed when Kill() is called so that Execute() can
-	// return even if cmd.Wait() is blocked. This is useful since cmd.Wait()
-	// sometimes does not return promptly.
-	abort chan struct{}
-}
+type (
+	Command struct {
+		ResourceMonitor func(chan *ResourceUsage, chan struct{})
+		mutex           sync.RWMutex
+		*exec.Cmd
+		// abort channel is closed when Kill() is called so that Execute() can
+		// return even if cmd.Wait() is blocked. This is useful since cmd.Wait()
+		// sometimes does not return promptly.
+		abort chan struct{}
+	}
 
-type Result struct {
-	SystemError error
-	ExitError   *exec.ExitError
-	Duration    time.Duration
-	Aborted     bool
-	KernelTime  time.Duration
-	UserTime    time.Duration
-}
+	Result struct {
+		SystemError error
+		ExitError   *exec.ExitError
+		Duration    time.Duration
+		Aborted     bool
+		KernelTime  time.Duration
+		UserTime    time.Duration
+		Usage       *ResourceUsage
+	}
+
+	ResourceUsage struct {
+		AverageMemoryUsed    uint64
+		PeakMemoryUsed       uint64
```

Comment:
Part of me feels like this should be AverageSystemMemoryUsed / PeakSystemMemoryUsed to make it clearer that it isn't process memory.